sprint3/zod.js

```diff
@@ -0,0 +1,49 @@
+import { z } from "zod";
+import isUuid from "is-uuid"
+
+const uuidSchema = z.string().refine(isUuid, {
+  message: "Invalid UUID format",
+});
+
+const CreateProduct = z.object({
+  id: uuidSchema,
+  name: z.string().min(2).max(100),
+  description: z.string().max(500).optional(),
+  price: z.number().min(0),
+  tags: z.array(z.string().min(2).max(100)),
+  createdAt: z.date(),
+  updatedAt: z.date(),
+});
+
+const CreateArticle = z.object({
+  id: uuidSchema,
+  title: z.string().min(2).max(100),
+  content: z.string().min(10).max(1000),
+  createdAt: z.date(),
+  updatedAt: z.date(),
+});
+
+const ProductComment = z.object({
+  id: uuidSchema,
+  content: z.string().min(10).max(1000),
+  createdAt: z.date(),
+  updatedAt: z.date(),
+  productId: uuidSchema,
+  articleId: uuidSchema,
+});
```

Comment:
이 유효성 검증기를 가지고, GET/POST에 동시 활용하고 있는 것 같네요.
GET 요청 시 데이터베이스의 응답 데이터 검증으로는 보통 사용하지 않아요.
PK, 시간 값은 보통 DB에서 자동으로 generated 해주기 때문이에요.
혹은 비즈니스 코드에서 생성 후 입력해준다고 하더라도, 입력 단에서 보통 검증하게 됩니다.

만약 DB에서 Default로 uuid, date값을 생성해주는 상황이라면 아래처럼 코드를 수정할 수 있습니다.
```suggestion
const CreateProductComment = z.object({
  content: z.string().min(1).max(1000)
  // id, createdAt, updatedAt은 DB에서 자동 생성
  // productComment에는 articleId가 없어야하고, productId도 검증하지 않음.
});
```